fuzz/fuzz_targets/fuzz.rs

```diff
@@ -390,23 +397,55 @@ impl<const D: usize> Workload<D> {
             // Due to sphere geometry, `Mode::Grid` doesn't imply traversals will agree.
             self.fuzz_traversal(&bvh, &flat_bvh, &self.ball.ball(), false);
 
+            let mut last_distance = f32::NEG_INFINITY;
             let nearest_traverse_iterator = bvh
                 .nearest_traverse_iterator(&ray, &self.shapes)
+                .inspect(|shape| {
+                    let distance = ray.intersection_slice_for_aabb(&shape.aabb()).unwrap().0;
+                    assert!(
+                        !assert_ray_traversal_agreement || distance >= last_distance,
+                        "{distance} {last_distance}"
+                    );
+                    last_distance = distance;
+                })
                 .map(ByPtr)
                 .collect::<HashSet<_>>();
+            let mut last_distance = f32::INFINITY;
             let farthest_traverse_iterator = bvh
                 .farthest_traverse_iterator(&ray, &self.shapes)
+                .inspect(|shape| {
+                    let distance = ray.intersection_slice_for_aabb(&shape.aabb()).unwrap().1;
+                    assert!(
+                        !assert_ray_traversal_agreement || distance <= last_distance,
+                        "{distance} {last_distance}"
+                    );
+                    last_distance = distance;
+                })
+                .map(ByPtr)
+                .collect::<HashSet<_>>();
+
+            let nearest_child_traverse_iterator = bvh
+                .nearest_child_traverse_iterator(&ray, &self.shapes)
+                .map(ByPtr)
+                .collect::<HashSet<_>>();
+            let farthest_child_traverse_iterator = bvh
+                .farthest_child_traverse_iterator(&ray, &self.shapes)
```

Comment:
This whole section is a bit sparse in terms of comments (but clearly this is not just boilerplate). Could you document the intent here?

Reply:
done :)